src/network/deployer.ts

```diff
@@ -0,0 +1,270 @@
+
+const IS_PACKAGED = Boolean((process as typeof process & {pkg?: unknown}).pkg);
+const PROXY_METADATA_KEY = 'zweb/contracts/proxy/metadata';
+
+// TODO: this is a temporary solution, supporting only limited number of dependencies
+const NPM_DEPS_URLS: Record<string, string> = {
+    '@openzeppelin/contracts': 'https://registry.npmjs.org/@openzeppelin/contracts/-/contracts-4.7.3.tgz',
+    '@openzeppelin/contracts-upgradeable': 'https://registry.npmjs.org/@openzeppelin/contracts-upgradeable/-/contracts-upgradeable-4.7.3.tgz'
+};
+
+const downloadNpmDependency = (dependency: string) => new Promise<void>(async (resolve, reject) => {
+    try {
+        log.debug({dependency}, 'Downloading npm dependency');
+        const depsPath = path.join(os.homedir(), '.point', 'hardhat', 'node_modules');
+        if (!fs.existsSync(depsPath)) {
+            await fs.mkdirp(depsPath);
+        }
+        
+        if (fs.existsSync(path.join(depsPath, dependency))) {
+            log.debug({dependency}, 'NPM dependency already exists, skipping');
+            resolve();
+        }
+
+        const res = await axios.get(NPM_DEPS_URLS[dependency], {responseType: 'stream'});
+
+        const stream = res.data
+            .pipe(gunzip())
+            .pipe(tar.extract(path.join(depsPath, `${dependency}_tmp`)));
+
+        stream.on('finish', async () => {
+            try {
+                if (dependency.split('/').length > 1) {
+                    await fs.mkdirp(path.join(depsPath, ...dependency.split('/').slice(0, -1)));
+                }
+                await fs.rename(
+                    path.join(depsPath, `${dependency}_tmp`, 'package'),
+                    path.join(depsPath, dependency)
+                );
+                await fs.rm(path.join(depsPath, `${dependency}_tmp`), {recursive: true, force: true});
+                log.debug({dependency}, 'Downloaded npm dependency successfully');
+                resolve();
+            } catch (e) {
+                reject(e);
+            }
+        });
+
+        stream.on('error', (e: Error) => {
+            reject(e);
+        });
+    } catch (e) {
+        reject(e);
+    }
+});
+
+const storeContractArtifacts = async ({
+    artifacts,
+    contractName,
+    version,
+    address,
+    target
+}: {
+    artifacts: Artifact,
+    contractName: string,
+    version: string,
+    address: string,
+    target: string
+}) => {
+    const artifactsJSON = JSON.stringify(artifacts);
+
+    const artifactsStorageId = await uploadFile(artifactsJSON);
+
+    await ethereum.putKeyValue(
+        target,
+        'zweb/contracts/address/' + contractName,
+        address,
+        version
+    );
+    await ethereum.putKeyValue(
+        target,
+        'zweb/contracts/abi/' + contractName,
+        artifactsStorageId,
+        version
+    );
+
+    return artifactsStorageId;
+};
+
+const getProxyMetadataFilePath = async () => {
+    const networkNames: Record<number, string> = {
+        1: 'mainnet',
```

Comment:
This should be added to config instead. Ideally it should be joined with the existing chain configurations.

Reply:
This is taken from the old deployer, I just didn't touch it. Though maybe it's simpler to remove completely